diff --git a/CMakeLists.txt b/CMakeLists.txt
index b4d11f3..e0fe6d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(lomiri-url-dispatcher C CXX)
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_lower) # Build types should always be lowercase but sometimes they are not.
 
diff --git a/cmake/UseConstantBuilder.cmake b/cmake/UseConstantBuilder.cmake
index 06c8d05..1dd9209 100644
--- a/cmake/UseConstantBuilder.cmake
+++ b/cmake/UseConstantBuilder.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
diff --git a/cmake/UseGlibGeneration.cmake b/cmake/UseGlibGeneration.cmake
index 67e34b9..bdd3156 100644
--- a/cmake/UseGlibGeneration.cmake
+++ b/cmake/UseGlibGeneration.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 if(POLICY CMP0011)
   cmake_policy(SET CMP0011 NEW)
 endif(POLICY CMP0011)
