diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70310a2..fc213d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,8 +89,10 @@ else()
 endif()
 message("Installing DBus interfaces to ${DBUSIFACEDIR}")
 
-pkg_check_modules(SYSTEMD REQUIRED systemd)
-pkg_get_variable(SYSTEMD_USER_UNIT_DIR systemd systemduserunitdir)
+pkg_check_modules(SYSTEMD systemd)
+if(SYSTEMD_FOUND)
+  pkg_get_variable(SYSTEMD_USER_UNIT_DIR systemd systemduserunitdir)
+endif()
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 06916c1..f3b9973 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -30,16 +30,18 @@ DESTINATION ${DBUSIFACEDIR}
 # systemd
 ###########################
 
-configure_file(
-lomiri-url-dispatcher.service.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
-)
-
-install(
-FILES
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
-DESTINATION ${SYSTEMD_USER_UNIT_DIR}
-)
+if(SYSTEMD_FOUND)
+  configure_file(
+  lomiri-url-dispatcher.service.in
+  ${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
+  )
+
+  install(
+  FILES
+  ${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
+  DESTINATION ${SYSTEMD_USER_UNIT_DIR}
+  )
+endif()
 
 ###########################
 # QML
