--- a/cargo_build.py	2021-07-03 00:25:41.337600000 +0300
+++ b/cargo_build.py	2021-07-03 00:26:47.498806285 +0300
@@ -9,14 +9,21 @@
 import shlex
 import subprocess
 import sys
+import os
 
 source_dir = Path(__file__).absolute().parent
 
 args = sys.argv[1:]
-binary_dir = "debug"
+if 'CROSS_BUILD' in os.environ:
+    binary_dir = os.environ['RUST_TARGET'] + "/debug"
+else:
+    binary_dir = "debug"
 
 if '--release' in args:
-    binary_dir = "release"
+    if 'CROSS_BUILD' in os.environ:
+        binary_dir = os.environ['RUST_TARGET'] + "/release"
+    else:
+        binary_dir = "release"
 
 # The file produced by Cargo will have a special name
 try:
