diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93c580dc..f83bef7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,9 @@ set(LDM_VERSION_MAJOR 0)
 set(LDM_VERSION_MINOR 1)
 set(LDM_VERSION_PATCH 0)
 
+# Options
+option(ENABLE_TESTS "Enable all tests and checks" ON)
+
 file(GLOB_RECURSE I18N_CPP_SRC_FILES
      RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
      src/**.cpp)
@@ -53,7 +56,6 @@ pkg_check_modules(DBUS REQUIRED dbus-1)
 pkg_check_modules(GLOG REQUIRED libglog)
 pkg_check_modules(LOMIRI_API REQUIRED liblomiri-api)
 
-enable_testing()
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pipe -std=c++11 -Werror -O2 -Wall -W -D_REENTRANT -fPIC -pedantic -Wextra")
 add_definitions("-DNDEBUG")
 
@@ -69,6 +71,9 @@ if(cmake_build_type_lower MATCHES relwithdebinfo) # workaround for http://pub
 endif()
 
 add_subdirectory(src)
-add_subdirectory(tests)
+if(ENABLE_TESTS)
+  enable_testing()
+  add_subdirectory(tests)
+endif()
 add_subdirectory(docs)
 add_subdirectory(po)
