diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ab56cc39..61dd05e7 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -14,7 +14,7 @@
 #
 # Authored by: Manuel de la Peña <manuel.delapena@canonical.com>
 
-find_package(GMock REQUIRED)
+find_package(GTest REQUIRED)
 include_directories(${GMOCK_INCLUDE_DIR} ${GTEST_INCLUDE_DIR})
 
 if(NOT "${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
diff --git a/tests/common/CMakeLists.txt b/tests/common/CMakeLists.txt
index 8e0611e8..5c58a47e 100644
--- a/tests/common/CMakeLists.txt
+++ b/tests/common/CMakeLists.txt
@@ -49,8 +49,7 @@ target_link_libraries(${TARGET}
     ${Qt5Sql_LIBRARIES}
     ${Qt5DBus_LIBRARIES}
     ${Qt5Test_LIBRARIES}
-    ${GMOCK_LIBRARIES}
-    ${GTEST_BOTH_LIBRARIES}
+    GTest::gmock_main
     ldm-common
     ldm-priv-common
     lomiri-download-manager-common
diff --git a/tests/downloads/daemon/CMakeLists.txt b/tests/downloads/daemon/CMakeLists.txt
index 90de49c5..ea1a430d 100644
--- a/tests/downloads/daemon/CMakeLists.txt
+++ b/tests/downloads/daemon/CMakeLists.txt
@@ -88,8 +88,7 @@ set(DAEMON_TESTS_LIBS
         ${Qt5Sql_LIBRARIES}
         ${Qt5DBus_LIBRARIES}
         ${Qt5Test_LIBRARIES}
-        ${GMOCK_LIBRARY}
-        ${GTEST_BOTH_LIBRARIES}
+        GTest::gmock_main
         ldm-common
         ldm-priv-common
         lomiri-download-manager-common
diff --git a/tests/downloads/qml/CMakeLists.txt b/tests/downloads/qml/CMakeLists.txt
index 3baa412c..a7f70f43 100644
--- a/tests/downloads/qml/CMakeLists.txt
+++ b/tests/downloads/qml/CMakeLists.txt
@@ -56,8 +56,7 @@ set(QML_TESTS_LIBS
     ${Qt5Sql_LIBRARIES}
     ${Qt5DBus_LIBRARIES}
     ${Qt5Test_LIBRARIES}
-    ${GMOCK_LIBRARY}
-    ${GTEST_BOTH_LIBRARIES}
+    GTest::gmock_main
     ldm-common
     lomiri-download-manager-common
     lomiri-download-manager-test-lib
diff --git a/tests/uploads/daemon/CMakeLists.txt b/tests/uploads/daemon/CMakeLists.txt
index 2637fad3..3be42bfc 100644
--- a/tests/uploads/daemon/CMakeLists.txt
+++ b/tests/uploads/daemon/CMakeLists.txt
@@ -55,8 +55,7 @@ set(UPLOAD_DAEMON_TESTS_LIBS
         ${Qt5Sql_LIBRARIES}
         ${Qt5DBus_LIBRARIES}
         ${Qt5Test_LIBRARIES}
-        ${GMOCK_LIBRARY}
-        ${GTEST_BOTH_LIBRARIES}
+        GTest::gmock_main
         ldm-common
         ldm-priv-common
         lomiri-upload-manager-common
