diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93e20fb..5f2d26f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(lomiri-settings-components)
-cmake_minimum_required(VERSION 2.8.9)
+cmake_minimum_required(VERSION 3.5)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
diff --git a/po/CMakeLists.txt b/po/CMakeLists.txt
index 60e37dc..ef89740 100644
--- a/po/CMakeLists.txt
+++ b/po/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.9)
+cmake_minimum_required(VERSION 3.5)
 find_package(Gettext REQUIRED)
 
 SET(POT_FILE
