diff --git a/CMakeLists.txt b/CMakeLists.txt
index 732f381..2192c7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,8 +83,10 @@ include_directories(${CLICK_INCLUDE_DIRS})
 pkg_check_modules(DBUS REQUIRED dbus-1)
 include_directories(${DBUS_INCLUDE_DIRS})
 
-pkg_check_modules(DBUSTEST REQUIRED dbustest-1>=14.04.0)
-include_directories(${DBUSTEST_INCLUDE_DIRS})
+if(enable_tests)
+  pkg_check_modules(DBUSTEST REQUIRED dbustest-1>=14.04.0)
+  include_directories(${DBUSTEST_INCLUDE_DIRS})
+endif()
 
 pkg_check_modules(LTTNG REQUIRED lttng-ust)
 include_directories(${LTTNG_INCLUDE_DIRS})
