From be3700a706d7bca56acc28105eb69c63f7008074 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Fri, 27 May 2022 17:17:11 +0300
Subject: [PATCH 3/8] CMake: Capitalize ENABLE_TESTS option

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index abc5c72..a2408a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set(ABI_VERSION 0)
 # Options
 ##########################
 
-option (enable_tests "Build tests" ON)
+option (ENABLE_TESTS "Build tests" ON)
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_MODULE_PATH}")
 
@@ -116,7 +116,7 @@ add_subdirectory(tools)
 add_subdirectory(utils)
 
 # testing & coverage
-if (${enable_tests})
+if (${ENABLE_TESTS})
   enable_testing ()
   add_subdirectory(tests)
 endif ()
-- 
2.36.1

